src/compute-client/src/controller.rs

```diff
@@ -515,72 +524,22 @@ where
     ComputeGrpcClient: ComputeClient<T>,
 {
     /// Adds replicas of an instance.
-    pub async fn add_replica_to_instance(
+    pub fn add_replica_to_instance(
```

Comment:
I noticed after rebasing that it's not possible anymore to have `remove_replica` async, because of the same reason you just mentioned.
Unfortunately I think that `update_read_capabilities` can even cause a orchestrator interaction...
```
src/storage/src/controller.rs:1304:           hosts.deprovision(id).await?;
```

Reply:
Ah, I had hoped the await there would only do something short-blocking, but you are right, it talks to the orchestrator.

I think we should talk to the storage team to move the deprovisioning of hosts out of `update_read_capabilities`. That looks like it would make a lot of methods non-async in both controllers. It could happen similarly to how rehydration works now in the compute controller, i.e. `update_read_capabilities` doesn't deprovision directly but adds hosts-to-remove to some list, and then in the next call to `process` it drains that list and deprovisions all hosts inside.